src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs

```diff
@@ -114,64 +112,53 @@ public PdbSourceDocumentMetadataAsSourceFileProvider(IPdbFileLocatorService pdbF
                 _assemblyToProjectMap.Add(assemblyName, projectId);
             }
 
+            var tempFilePath = Path.Combine(tempPath, projectId.Id.ToString());
+            // Create the directory. It's possible a parallel deletion is happening in another process, so we may have
+            // to retry this a few times.
+            var loopCount = 0;
+            while (!Directory.Exists(tempFilePath))
+            {
+                // Protect against infinite loops.
+                if (loopCount++ > 10)
+                    return null;
+
+                try
+                {
+                    Directory.CreateDirectory(tempFilePath);
+                }
+                catch (DirectoryNotFoundException)
```

Comment:
❓Is there any specific reason this doesn't just use `IOUtilities.PerformIO`?

Reply:
No, good call. This was just lazily copied from the decompilation source provider